dashboard/generate_subnational_map.R

```diff
@@ -7,71 +7,86 @@ library(sf)
 library(tmap)
 
 # read in data to display
-ipc_indices_data <- read.csv("Nov2020_FewsNetGlobalIPC.csv")
+ipc_indices_data <- read.csv("data/foodinsecurity/ethiopia_foodinsec_trigger.csv") #Nov2020_FewsNetGlobalIPC.csv")
+#last date of fewsnet and global ipc can differ, so select them separately and combine them
+ipc_indices_data_latest_fn <- ipc_indices_data %>% filter(source == "FewsNet") %>% filter(date == max(date))
+ipc_indices_data_latest_gbl <- ipc_indices_data %>% filter(source == "GlobalIPC") %>% filter(date == max(date))
+ipc_indices_data_latest <- rbind(ipc_indices_data_latest_fn,ipc_indices_data_latest_gbl)
 
 # import shapefiles
-eth_adm1 <- st_read("data/shapefiles/ET_Admin1_OCHA_2019/eth_admbnda_adm1_csa_bofed_20190827.shp", stringsAsFactors = F)
+eth_adm1 <- st_read("data/shapefiles/ET_Admin_OCHA_2020/eth_admbnda_adm1_csa_bofed_20201008.shp", stringsAsFactors = F)
 
 # build datasets per country, source. Ensures all regions are represented if no projections in certain regions
 # ETH FN
 eth_fn <-  eth_adm1 %>%
-             left_join(ipc_indices_data[ipc_indices_data$Source == 'FewsNet',], by = c('ADM1_EN' = 'ADMIN1'))
+             left_join(ipc_indices_data_latest_fn, by = c('ADM1_EN' = 'ADMIN1'))
+
+# ETH GLB  
+# For global IPC, it can be the case that for some dates, some admins do have ML1 projections but no ML2.
+# In this case we want to show those admins as missing in the ML2 map
+# Therefore, only select the admins with ML1/ML2 projections (indicated by a not-nan ML1/ML2 population)
+eth_gbl_ml1 <-  eth_adm1 %>%
+              left_join(ipc_indices_data_latest_gbl %>% drop_na(pop_ML1), by = c('ADM1_EN' = 'ADMIN1'))
```

Comment:
why would we want to drop a whole row if it only has one set of projections? And if we do, shouldn't we drop_na(pop_ML2) since these are the projections that are sometimes missing?

Reply:
Ok I see you address missing ML2 in the next few lines. Are there cases where we lack ML1s?

Reply:
I haven't seen cases where we lack ML1s but do have ML2s, but I can imagine they will appear in the future thus prepared the code for that

Reply:
And so the row is not dropped if only one of the two is missing, separate dataframes are made for ML1 and ML2 which result in separate maps and trigger lists.